### DISCUSSION TITLE
Wrong base href URL. Is it possible to set domain url for Authelia?

### DISCUSSION THREAD

**Discussion by ilu123** [authelia/authelia#5432] [Q&A] [answered]
I setup the Authelia with Nginx, and pass the requests to authelia by
```
proxy_pass http://authelia:9091;
```
My domain name is ` xxx.yyy.com`.

The login page and **RegisterDevice** mail sending are working well, but the URL in **mail** and the **one-time-password** page's `base href=URL` are both wrong, which all becomes` "http://authelia:9091"`.
I tested the mail link by changing it to` "https://xxx.yyy.com/one-time-password/......."`,  but still got errors as described in SS.

I checked all the docs about configurations but not found out how to make that to my domain name and HTTPS protocol. 

So please help. Thanks.

SS:
![image](https://github.com/authelia/authelia/assets/33821966/a61ade0b-5402-450c-9b73-384bfcb6dc23)

**Comment by james-d-elliott** [comment, marked as answer]
https://www.authelia.com/integration/proxies/introduction/#required-headers

**Reply by ilu123** [reply]
@james-d-elliott  thanks, well, I also set those headers  even hardcoded, but still not working at all....

**Reply by james-d-elliott** [reply]
Double check it, we test this behavior as part of our integration testing with 6 different proxies.

**Reply by ilu123** [reply]
ok, well, what should be modified is proxy.conf part, right? here is mine:
```
...
## If the subreqest returns 200 pass to the backend, if the subrequest returns 401 redirect to the portal.
error_page 401 =302 https://$http_host/login/?rd=$target_url;
[root nginx_config]# cat authelia_proxy.conf
## Headers
proxy_set_header Host xxx.yyy.com;
proxy_set_header X-Original-URL $scheme://$http_host$request_uri;
proxy_set_header X-Forwarded-Proto https; #$scheme; <=================================Hardcode
proxy_set_header X-Forwarded-Host xxx.yyy.com; #$http_host; <=================================Hardcode
proxy_set_header X-Forwarded-Uri $request_uri;
proxy_set_header X-Forwarded-Ssl on;
proxy_set_header X-Forwarded-For $remote_addr;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header Connection "";
...
```

**Reply by james-d-elliott** [reply]
It doesn't matter which file, just matters that it's being applied via explicit directives (in the authelia server) or via implicit directives (by having an include in the authelia server that includes the snippet). You should confirm it using a container which prints headers like traefik/whoami. Or I believe log level trace will show it.

**Reply by ilu123** [reply]
I see....I will continue to check that then.
btw, if it has an env var to set its domain no matter xforwardedhost is, would that be more helpful? (sorry am not quite undertand the logic).
And one more question, I saw the authelia is startup with path "/", is it possible to let it run with a relative url, such as "/authelia" ?

**Reply by james-d-elliott** [reply]
There are no plans to hardcode this in this way. We have implemented several things that require these headers to make decisions. For example when supporting multiple cookie domains.

That's what the server path config is for: https://www.authelia.com/configuration/miscellaneous/server/#path

**Reply by ilu123** [reply]
Ah, understood!

How could I missed that `path:` !

Thanks a lot for your help!!!

**Reply by james-d-elliott** [reply]
No worries, it's a fairly uncommon config option to use (most people  use  subdomains). I'll keep it in mind for the FAQ
